flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkSqlParserImplTest.java

```diff
@@ -611,6 +611,70 @@ void testAlterTableAddWatermark() {
                 .fails("Multiple WATERMARK statements is not supported yet.");
     }
 
+    @Test
+    void testAlterTableAddDistribution() {
+        sql("alter table t1 add (DISTRIBUTION BY HASH(a, h) INTO 6 BUCKETS\n)")
+                .ok("ALTER TABLE `T1` ADD (\n  DISTRIBUTION BY HASH(`A`, `H`) INTO 6 BUCKETS\n)");
+
+        sql("alter table t1 add DISTRIBUTION BY HASH(a, h) INTO 6 BUCKETS\n")
+                .ok("ALTER TABLE `T1` ADD (\n  DISTRIBUTION BY HASH(`A`, `H`) INTO 6 BUCKETS\n)");
+
+        sql("alter table tbl1 add DISTRIBUTION BY HASH(a, h) INTO 6 BUCKETS\n")
+                .ok("ALTER TABLE `TBL1` ADD (\n  DISTRIBUTION BY HASH(`A`, `H`) INTO 6 BUCKETS\n)");
+
+        sql("alter table tbl1 add DISTRIBUTION BY RANGE(a, h) INTO 6 BUCKETS\n")
+                .ok(
+                        "ALTER TABLE `TBL1` ADD (\n  DISTRIBUTION BY RANGE(`A`, `H`) INTO 6 BUCKETS\n)");
+
+        sql("alter table tbl1 add DISTRIBUTION BY ^RANDOM^(a, h) INTO 6 BUCKETS\n")
+                .fails("(?s).*Encountered \"RANDOM\" at line 1, column 38.*");
+
+        sql("alter table tbl1 add DISTRIBUTION BY (a, h) INTO 6 BUCKETS\n")
+                .ok("ALTER TABLE `TBL1` ADD (\n  DISTRIBUTION BY (`A`, `H`) INTO 6 BUCKETS\n)");
+
+        sql("alter table tbl1 add DISTRIBUTION BY RANGE(a, h)\n")
+                .ok("ALTER TABLE `TBL1` ADD (\n  DISTRIBUTION BY RANGE(`A`, `H`)\n)");
+
+        sql("alter table tbl1 add DISTRIBUTION BY (a, h)\n")
+                .ok("ALTER TABLE `TBL1` ADD (\n  DISTRIBUTION BY (`A`, `H`)\n)");
+    }
+
+    @Test
+    void testAlterTableModifyDistribution() {
+        sql("alter table t1 modify (DISTRIBUTION BY HASH(a, h) INTO 6 BUCKETS\n)")
```

Comment:
Same comment as above. Parenthesis should not be allowed. Distribution is not a schema component. It deserves its own branch.